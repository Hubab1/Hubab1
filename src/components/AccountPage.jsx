import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Formik } from 'formik';
import * as Yup from 'yup';
import Grid from '@material-ui/core/Grid';
import { KeyboardDatePicker } from '@material-ui/pickers';

import { updateApplicant } from 'reducers/applicant';
import captureRoute from 'app/captureRoute';
import { H1, formContent } from 'assets/styles';
import FormTextInput from 'components/common/FormTextInput/FormTextInput';
import PhoneNumberInput from 'components/common/PhoneNumberInput';
import GenericFormError from 'components/common/GenericFormError';
import ActionButton from 'components/common/ActionButton/ActionButton';
import { ROUTES } from 'app/constants';
import { allValuesSet } from 'utils/formik';
import { serializeDate, parseDateISOString } from 'utils/misc';	


export class AccountPage extends React.Component {
    state = {errors: null}

    get initialValues () {
        const applicant = this.props.applicant;
        let birthday = applicant.birthday;
        if (birthday) {
            birthday = parseDateISOString(birthday);
        }
        return {
            first_name: applicant.first_name,
            last_name: applicant.last_name,
            phone_number: applicant.phone_number,
            email: applicant.email,
            birthday
        }
    }

    onSubmit = (values, { setSubmitting, setErrors }) => {
        const stateUpdate = Object.assign({}, values);
        stateUpdate.birthday = serializeDate(stateUpdate.birthday);
        this.props.updateApplicant(stateUpdate, stateUpdate).then((res) => {
            if (res.errors) {
                setErrors(res.errors);
            } else {
                
            }
            setSubmitting(false);
        });
    }

    render () {
        return (
            <>
                <H1>Your Account Details</H1>
                <Formik
                    initialValues={this.initialValues}
                    validationSchema={Yup.object().shape({
                        first_name: Yup.string().required('First Name is required'),
                        last_name: Yup.string().required('Last Name is required'),
                        phone_number: Yup.string()
                            .required('Phone Number is required')
                            .matches(/^\(\d{3}\)\s\d{3}-\d{4}/, 'Must be a valid US phone number'),
                        email: Yup.string()
                            .email()
                            .required('Email is required'),
                        birthday: Yup.string()	
                            .required('required'),
                    })}
                    onSubmit={this.onSubmit}
                >
                    {({
                        values,
                        errors,
                        handleChange,
                        submitCount,
                        handleBlur,
                        handleSubmit,
                        isSubmitting,
                        touched,
                        setFieldValue,
                    }) => (
                        <form onSubmit={handleSubmit} autoComplete="off">
                            <div className={formContent}>
                                { this.state.errors && <GenericFormError errors={this.state.errors}/> }
                                <Grid container spacing={1}>
                                    <Grid item xs={12}>
                                        <FormTextInput
                                            label="First Name"
                                            name="first_name"
                                            submitted={submitCount > 0}
                                            handleChange={handleChange}
                                            handleBlur={handleBlur}
                                            error={errors.first_name}
                                            value={values.first_name}
                                        />
                                    </Grid>
                                    <Grid item xs={12}>
                                        <FormTextInput
                                            label="Last Name"
                                            name="last_name"
                                            submitted={submitCount > 0}
                                            handleChange={handleChange}
                                            handleBlur={handleBlur}
                                            error={errors.last_name}
                                            value={values.last_name}
                                        />
                                    </Grid>
                                    <Grid item xs={12}>
                                        <FormTextInput
                                            label="Email"
                                            name="email"
                                            submitted={submitCount > 0}
                                            handleChange={handleChange}
                                            handleBlur={handleBlur}
                                            error={errors.email}
                                            value={values.email}
                                        />
                                    </Grid>
                                    <Grid item xs={6}>
                                        <PhoneNumberInput 
                                            label="Phone Number"
                                            name="phone_number"
                                            value={values.phone_number}
                                            handleChange={handleChange}
                                            error={submitCount > 0 && !!errors.phone_number}
                                            helperText={submitCount > 0 ? errors.phone_number : null}
                                        />
                                    </Grid>
                                    <Grid item xs={6}>
                                        <KeyboardDatePicker	
                                            id="birthday-picker"	
                                            clearable	
                                            disableFuture	
                                            format="MM/dd/yyyy"	
                                            placeholder="mm/dd/yyyy"	
                                            label="Birthday"	
                                            value={values.birthday || null}	
                                            fullWidth	
                                            onBlur={handleBlur}	
                                            onChange={e => setFieldValue('birthday', e)}	
                                            KeyboardButtonProps={{	
                                                'aria-label': 'change date',	
                                            }}	
                                        />	
                                    </Grid>
                                </Grid>
                                <ActionButton disabled={!allValuesSet(values) || isSubmitting} marginTop={20} marginBottom={20}>Save Changes</ActionButton>
                            </div>
                        </form>
                    )}
                </Formik>
            </>
        );
    }
}

AccountPage.propTypes = {
    updateApplicant: PropTypes.func,
}

const mapStateToProps = (state) => ({
    applicant: state.applicant,
});

const mapDispatchToProps = { updateApplicant };

export default connect(mapStateToProps, mapDispatchToProps)(captureRoute(AccountPage, ROUTES.ACCOUNT));
