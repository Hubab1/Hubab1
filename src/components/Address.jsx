import React, { useState, useCallback, useMemo } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { TextField } from 'formik-material-ui';
import { Formik, Form, Field } from 'formik';
import * as Yup from 'yup';
import styled from '@emotion/styled';

import { ROUTES } from 'app/constants';
import { updateApplicant } from 'reducers/applicant';
import withRelativeRoutes from 'app/withRelativeRoutes';
import ActionButton from 'components/common/ActionButton/ActionButton';
import LocationSearch from 'components/common/Formik/LocationSearch/LocationSearch';
import { H1, SpacedH3 } from 'assets/styles';
import sticky from 'assets/images/sticky.png';
import GenericFormMessage from './common/GenericFormMessage';

const ImageContainer = styled.div`
    margin-top: 31px;
    margin-bottom: 31px;
    img {
        max-height: 105px;
        max-width: 114px;
    }
`;

export const validationSchema = Yup.object().shape({
    address_search: Yup.object().shape({
        address_street: Yup.string()
            .required('Street is required')
            .matches(/^[A-Za-z0-9]+(?:\s[A-Za-z0-9'_-]+)+$/, 'Invalid street'),
        address_city: Yup.string()
            .required('City is required')
            .matches(/^[\w'\-,.][^0-9_!¡?÷?¿/\\+=@#$%ˆ&*(){}|~<>;:[\]]{1,}$/, 'Invalid city'),
        address_state: Yup.string().required('State is required'),
        address_postal_code: Yup.number().required('Zip code is required'),
    }),
    address_line_2: Yup.string(),
});

export const Address = ({ applicant, updateApplicant, _nextRoute }) => {
    const [errors, setErrors] = useState(null);
    const handleSubmit = useCallback(
        async (values, { setSubmitting, setErrors: setFormErrors }) => {
            try {
                const serialized = {
                    ...values.address_search,
                    address_line_2: values.address_line_2,
                };

                const response = await updateApplicant(serialized);
                if (response.errors) {
                    setFormErrors(response.errors);
                } else {
                    _nextRoute();
                }
            } catch {
                setErrors(['Oops! We ran into some issues. Please try again later.']);
            } finally {
                setSubmitting(false);
            }
        },
        [updateApplicant, _nextRoute]
    );

    const initialValues = useMemo(() => {
        const searchBuilder = [];
        if (applicant.address_street) searchBuilder.push(applicant.address_street);
        if (applicant.address_city) searchBuilder.push(applicant.address_city);
        if (applicant.address_state) searchBuilder.push(applicant.address_state);
        if (applicant.address_postal_code) searchBuilder.push(applicant.address_postal_code);
        const search = searchBuilder.join(', ');

        return {
            address_search: {
                search,
                address_street: applicant.address_street,
                address_city: applicant.address_city,
                address_state: applicant.address_state,
                address_postal_code: applicant.address_postal_code,
            },
            address_line_2: applicant.address_line_2,
        };
    }, [applicant]);

    if (!applicant) {
        return null;
    }

    return (
        <>
            <H1>Tell Us A Little More</H1>
            <SpacedH3>Now, by filling out these details below we can screen you more accurately.</SpacedH3>
            <ImageContainer>
                <img src={sticky} alt="sticky note" />
            </ImageContainer>
            <Formik validationSchema={validationSchema} initialValues={initialValues} onSubmit={handleSubmit}>
                {({ values, isSubmitting, handleSubmit, submitCount }) => {
                    const disableSubmit = !values.address_search.search || isSubmitting;

                    return (
                        <Form onSubmit={handleSubmit} autoComplete="off">
                            <Field
                                fullWidth
                                margin="normal"
                                label="Street name, city, state, zip"
                                name="address_search"
                                submitCount={submitCount}
                                component={LocationSearch}
                            />
                            <Field
                                fullWidth
                                margin="normal"
                                label="Apt/Ste/Floor"
                                name="address_line_2"
                                component={TextField}
                            />
                            <GenericFormMessage type="error" messages={errors} />
                            <ActionButton marginTop={50} disabled={disableSubmit}>
                                Continue
                            </ActionButton>
                        </Form>
                    );
                }}
            </Formik>
        </>
    );
};

Address.propTypes = {
    updateApplicant: PropTypes.func.isRequired,
    applicant: PropTypes.object,
    _nextRoute: PropTypes.func,
};

const mapStateToProps = (state) => ({
    applicant: state.applicant,
});

const mapActionsToProps = {
    updateApplicant,
};

export default connect(mapStateToProps, mapActionsToProps)(withRelativeRoutes(Address, ROUTES.ADDRESS));
