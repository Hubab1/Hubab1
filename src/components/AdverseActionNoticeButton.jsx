import React, { useState } from 'react';
import API from 'app/api';
import { LinkButton } from 'assets/styles';
import { css } from 'emotion';

export const aanLink = css`
    outline: none;
`;

export function AdverseActionNoticeButton() {
    const [error, setError] = useState(false);

    const openDocument = async () => {
        const w = window.open('data:text/plain;charset=utf-8;base64,TG9hZGluZw==', '_blank');
        try {
            const response = await API.fetchAANDocument();
            const blobUrl = URL.createObjectURL(new Blob([response], { type: 'application/pdf' }));
            w.location.replace(blobUrl);
        } catch (e) {
            setError(true);
        }
    };

    return (
        <LinkButton onClick={openDocument} className={aanLink}>
            {error ? 'An error occurred. Please try again.' : 'Learn why'}
        </LinkButton>
    );
}
