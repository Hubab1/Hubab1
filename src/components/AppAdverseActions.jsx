import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

import ActionButton from 'components/common/ActionButton/ActionButton';
import { H1, P, Card, CardSection, CardRow, Bold } from 'assets/styles';
import API from 'app/api';
import { connect } from 'react-redux';

export const Subtitle = styled.small`
    color: #818797;
    font-size: 18px;
    line-height: 22px;
    text-align: center;
`;

const Header = styled.div`
    border-bottom: 1px solid #eeeeee;
    padding: 20px;
`;

const CreditScore = styled.div`
    width: 48px;
    height: 48px;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    font-size: 22px;
    border: 1px solid;
    font-weight: bold;
    color: #828796;
    color: ${(props) => props.color || '#828796'};
    border-color: ${(props) => props.color || '#828796'};
    margin-left: 1px;
    margin-right: 9px;
`;

const CreditScoreDetails = styled.div`
    padding-top: 3px;
`;

const Decision = styled.div`
    margin-top: 13px;
    white-space: pre-line;
    text-align: left;
    overflow-wrap: normal;
`;

const AdverseActionsFactors = styled.ul`
    padding-left: 16px;
    margin-top: 5px;
    margin-bottom: 0;
    li {
        font-size: 16px;
        letter-spacing: 0;
        line-height: 20px;
        margin-bottom: 12px;
    }
    li:last-child {
        margin-bottom: 0;
    }
`;

const IndentedRow = styled(CardRow)`
    justify-content: initial;
`;

export function AppAdverseActions(props) {
    const { configuration } = props;
    const [adverseFactors, setAdverseFactors] = useState([]);
    const [requestDate, setRequestDate] = useState(null);
    const [creditScore, setCreditScore] = useState('N/A');
    const [isReady, setIsReady] = useState(false);

    useEffect(() => {
        API.getAdverseActions()
            .then((res) => {
                if (res.adverse_factors) setAdverseFactors(res.adverse_factors);
                if (res.request_date) {
                    const formattedDate = new Date(res.request_date);
                    setRequestDate(
                        formattedDate.toLocaleDateString('en-US', { year: 'numeric', month: '2-digit', day: '2-digit' })
                    );
                }
                if (res.credit_score) setCreditScore(res.credit_score);
                setIsReady(true);
            })
            .catch(() => {
                setIsReady(true);
            });
    }, []);

    const getButtonText = () => {
        if (isReady) return 'OK';
        return 'Loading...';
    };

    const formatAdverseActionDate = (date) => {
        const d = new Date(date);
        return d.toLocaleDateString('en-US', { year: 'numeric', month: '2-digit', day: '2-digit' });
    };

    const getCreditScoreColor = (score) => {
        const ratingConfig = configuration.credit_score_rating_config;

        if (!ratingConfig || !ratingConfig.length || !score || isNaN(score)) return '#828796';

        let index = 0;

        for (let i = ratingConfig.length - 1; i >= 0; i--) {
            if (score >= ratingConfig[i][0]) break;
            index++;
        }

        switch (index) {
            case 0:
                return '#60C28D';
            case 1:
                return '#FCC022';
            default:
                return '#D0021B';
        }
    };

    const guarantor_income_requirement_multiplier = configuration.guarantor_income_requirement_multiplier;

    return (
        <div>
            <Header>
                <H1>Reason for Our Decision</H1>
                <br />
                <Subtitle>Adverse action notice</Subtitle>
            </Header>
            {isReady && (
                <Decision>
                    <P color="#818797">{formatAdverseActionDate(props.date)}</P>
                    <br />
                    <P>Dear {props.name},</P>
                    <br />
                    <P>
                        Thank you for your recent rental application for {props.buildingName}
                        {props.buildingName && props.unitNumber && <>,</>}
                        {props.unitNumber}.
                    </P>
                    <br />
                    {props.securityDeposit && !props.guarantorRequested && (
                        <>
                            {props.hasInternationalApplicant ? (
                                <P>
                                    Unfortunately, we are unable to approve your rental application under our standard
                                    terms and conditions because there is an international applicant.
                                </P>
                            ) : (
                                <P>
                                    Unfortunately, we are unable to approve your rental application under our standard
                                    terms and conditions.
                                </P>
                            )}
                            <br />
                            <P>
                                We can, however, approve your rental application if you agree to pay a security deposit
                                in the amount of {props.securityDeposit}, due at move in.
                            </P>
                        </>
                    )}

                    {props.guarantorRequested && (
                        <>
                            <P>
                                Unfortunately, we are unable to approve your rental application under our standard terms
                                and conditions because you do not meet the required household income.
                            </P>
                            <br />
                            <P>
                                {`We can, however, reevaluate your rental application if you add a guarantor. Guarantors are required to make ${guarantor_income_requirement_multiplier}x the monthly rent.`}
                            </P>
                        </>
                    )}
                    {!props.securityDeposit && !props.guarantorRequested && (
                        <P>Unfortunately, we are unable to approve your rental application.</P>
                    )}

                    <br />
                    {adverseFactors && adverseFactors.length > 0 && (
                        <Card>
                            <CardSection>
                                <CardRow>
                                    <P>Factors affecting our decision:</P>
                                </CardRow>
                                <CardRow>
                                    <AdverseActionsFactors>
                                        {adverseFactors.map((factor) => (
                                            <li key={factor}>{factor}</li>
                                        ))}
                                    </AdverseActionsFactors>
                                </CardRow>
                            </CardSection>
                        </Card>
                    )}
                    {requestDate && (
                        <Card>
                            <CardSection>
                                <IndentedRow>
                                    <CreditScore color={getCreditScoreColor(creditScore)}>{creditScore}</CreditScore>
                                    <CreditScoreDetails>
                                        <P>Your credit score as of {requestDate}</P>
                                        <P color="#818797" fontSize={14}>
                                            This score has a range of 350 to 850
                                        </P>
                                    </CreditScoreDetails>
                                </IndentedRow>
                            </CardSection>
                        </Card>
                    )}
                    <P>
                        In evaluating your application, <Bold fontWeight={600}>TransUnion Consumer Solutions</Bold>{' '}
                        provided us with information that in whole or in part influenced our decision.{' '}
                        <Bold fontWeight={600}>
                            The reporting agency played no part in our decision other than providing us with information
                            about you.
                        </Bold>
                    </P>
                    <br />
                    <P>
                        Under the Fair Credit Reporting Act, you have the right to know the information provided to us,
                        which can be obtained by contacting:
                    </P>
                    <br />
                    <P>
                        <Bold fontWeight={600}>TransUnion Consumer Solutions</Bold>
                        <br />
                        P.O. Box 2000
                        <br />
                        Chester, PA. 19022-2000
                        <br />
                        <a href="tel:1-800-916-8800">1-800-916-8800</a>
                    </P>
                    <br />
                    <P>
                        Background Data Solutions
                        <br />
                        <a href="tel:1-800-568-5665">1-800-568-5665</a>
                        <br />
                    </P>
                    <br />
                    <P>
                        You have the right to a free copy of your report from the reporting agency, if you request it no
                        later than 60 days after you receive this notice. In addition, if you find that any information
                        contained in the report you receive is inaccurate or incomplete, you have the right to dispute
                        the matter with the reporting agency that provided the report or through{' '}
                        <Bold fontWeight={600}>TransUnion Rental Screening Solutions</Bold> consumer relations at{' '}
                        <a href="tel:1-800-230-9376">1-800-230-9376</a> or{' '}
                        <a href="mailto:consumer_relations@turss.com">consumer_relations@turss.com</a>.
                    </P>
                    <br />
                </Decision>
            )}
            <ActionButton disabled={!isReady} marginTop={39} onClick={props.onAgree}>
                {getButtonText()}
            </ActionButton>
        </div>
    );
}
AppAdverseActions.propTypes = {
    date: PropTypes.string,
    buildingName: PropTypes.string,
    unitNumber: PropTypes.string,
    name: PropTypes.string,
    securityDeposit: PropTypes.string,
    onAgree: PropTypes.func.isRequired,
    configuration: PropTypes.object,
    guarantorRequested: PropTypes.bool,
    hasInternationalApplicant: PropTypes.bool,
};

const mapStateToProps = (state) => ({
    configuration: state.configuration,
});

export default connect(mapStateToProps)(AppAdverseActions);
