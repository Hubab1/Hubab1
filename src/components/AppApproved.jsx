import React from 'react';
import HelloSign from 'hellosign-embedded';
import styled from '@emotion/styled';
import Grid from '@material-ui/core/Grid';
import { css } from 'emotion';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import API from 'app/api';
import { ROUTES, HELLOSIGN_TEST_MODE, HELLOSIGN_CLIENT_ID, MILESTONE_LEASE_SENT } from 'app/constants';
import withRelativeRoutes from 'app/withRelativeRoutes';
import approvedSign from 'assets/images/approvedSign.svg';
import { H1, leftText, SpacedH3 } from 'assets/styles';
import ActionButton from 'components/common/ActionButton/ActionButton';
import lightbulb from 'assets/images/lightbulb.png';
import { prettyCurrency } from 'utils/misc';

export const ApprovedImage = styled.img`
    padding-top: 10px;
    height: 200px;
`;

export const applicationUnit = css`
  color: #454B57;
  font-size: 14px;
  line-height: 17px;
  text-align: center;
  padding-top: 10px;
`;

export const BulbImage = styled.img`
  width: 46px;
  height: 42px;
`;

export const securityDepositHelpText = css`
  color: #454B57;
  font-size: 14px;
  line-height: 17px;
  text-align: left;
`;

export const gridContainer = css`
    padding-top: 35px;
    min-height: 100px;
`;

export const AppApproved = ({profile, configuration, history}) => {
    if (!profile || ! configuration) return null;

    const {
        unit,
        security_deposit: securityDeposit,
        security_deposit_multiplier: securityDepositMultiplier,
    } = profile;
    const buildingName = configuration.community.building_name || configuration.community.normalized_street_address;
    const unitNumber = (!!unit && !!unit.unit_number) ? ` Unit ${unit.unit_number}` : '';

    const openEmbeddedSigning = async () => {
        const client = new HelloSign({
            clientId: HELLOSIGN_CLIENT_ID,
        });
        const data = await API.embeddedSigningUrl();
        if (data.url) {
            client && client.open(data.url, {
                testMode: HELLOSIGN_TEST_MODE,
                skipDomainVerification: HELLOSIGN_TEST_MODE,
            });
        }
        client.on('sign', () => {
            history.push(ROUTES.LEASE_SIGNED);
        });
    }
    const leaseSent = !!profile.events.find(e => String(e.event) === String(MILESTONE_LEASE_SENT));

    return (
        <>
            <H1>You've Been Approved!</H1>
            {
                leaseSent && <SpacedH3>All that's left to do is sign the lease.</SpacedH3>
            }
            {
                !leaseSent && <SpacedH3>We'll send an email with instructions on how to sign the lease shortly.</SpacedH3>
            }
            <ApprovedImage src={approvedSign}/>
            <div id="application-unit" className={applicationUnit}>{buildingName}{unitNumber}</div>
            <div className={gridContainer}>
                {securityDeposit &&
                <Grid container justify={'center'} className="security-deposit-container">
                    <Grid item xs={2}>
                        <BulbImage alt="light bulb" src={lightbulb} />
                    </Grid>
                    <Grid item xs={9} classes={{ root: leftText }}>
                        <span className={securityDepositHelpText}>
                            You've been approved under the condition that you agree to
                            a {prettyCurrency(securityDeposit)} security deposit
                            ({securityDepositMultiplier}x the rent).
                        </span>
                    </Grid>
                </Grid>}
                {
                    leaseSent &&
                    <ActionButton onClick={openEmbeddedSigning} marginTop={securityDeposit ? 30 : 90}>
                        Review &amp; Sign Lease
                    </ActionButton>
                }
            </div>
        </>
    )
};

AppApproved.propTypes = {
    profile: PropTypes.object,
    configuration: PropTypes.object,
};


const mapStateToProps = state => ({
    profile: state.renterProfile,
    configuration: state.configuration,
});

export default connect(mapStateToProps)(withRelativeRoutes(AppApproved, ROUTES.APP_APPROVED));
