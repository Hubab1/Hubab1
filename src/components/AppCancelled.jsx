import React from 'react';
import styled from '@emotion/styled';
import { css } from 'emotion';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { ROUTES } from 'app/constants';
import { H1, SpacedH3, P } from 'assets/styles';
import cancelled from 'assets/images/cancelled.png';
import { prettyFormatPhoneNumber } from 'utils/misc';
import captureRoute from 'app/captureRoute';

export const Img = styled.img`
    padding-top: 10px;
    height: 200px;
`;

export const applicationUnit = css`
  color: #454B57;
  font-size: 14px;
  line-height: 17px;
  text-align: center;
  padding-top: 10px;
`;

export const AppCancelled = ({profile, configuration}) => {
    if (!profile || ! configuration) return null;

    return (
        <>
            <H1>Application Cancelled</H1>
            <SpacedH3>You are no longer applying for unit {profile.unit.unit_number}.</SpacedH3>
            <Img src={cancelled}/>
            <P margin="90px 0 0 0" bold>Call us at <a href={`tel:${configuration.community.contact_phone}`}>{prettyFormatPhoneNumber(configuration.community.contact_phone)}</a>
                &nbsp;if you have any questions or if youâ€™d like to start a new application.
            </P>
        </>
    )
};

AppCancelled.propTypes = {
    profile: PropTypes.object,
    configuration: PropTypes.object,
};


const mapStateToProps = state => ({
    profile: state.renterProfile,
    configuration: state.configuration,
});

export default connect(mapStateToProps)(captureRoute(AppCancelled, ROUTES.APP_CANCELLED));
