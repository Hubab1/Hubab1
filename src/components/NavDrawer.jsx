import React, { useContext } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import CssBaseline from '@material-ui/core/CssBaseline';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Box from '@material-ui/core/Box';
import { Link } from 'react-router-dom';

import { Bold } from 'assets/styles';
import ProgressBar from 'components/common/Page/ProgressBar';
import { AppTheme } from 'contexts/AppContextProvider';
import BannerLogo from 'components/common/Page/BannerLogo';
import { drawerContent } from 'components/common/Page/styles';
import UnauthenticatedPage from 'components/common/Page/UnauthenticatedPage';
import NavStepper from './NavStepper';
import { ROUTES } from 'app/constants';

const useStyles = makeStyles(theme => ({
    list: {
        width: 300,
    },
    padRight: {
        width: 48
    },
    drawerHeader: {
        display: 'flex',
        alignItems: 'center',
        padding: '0 8px',
        ...theme.mixins.toolbar,
        justifyContent: 'flex-end',
    },
    darkThemeAppbar: {
        backgroundColor: theme.palette.primary.main,
        color: theme.palette.primary.contrastText
    },
    liteThemeAppbar: {
        backgroundColor: '#ffffff',
        color: '#000000'
    },
    toolbar: { 
        minHeight: 76
    },
    accountDetails: {
        padding: '28px 15px 15px 15px',
        borderBottom: '1px solid #EEEEEE',
    },
    initialsContainer: {
        color: '#828796',
        backgroundColor: '#EFEFEF',
        borderRadius: 50,
        textAlign: 'center',
        fontSize: 17,
        height: 40,
        width: 40,
        lineHeight: '40px',
        margin: '0 10px 10px 0',
    }
}));


export function PersistentDrawerLeft(props) {
    if (props.hideNav) {
        return <UnauthenticatedPage>{props.children}</UnauthenticatedPage>
    }
    const appThemeContext = useContext(AppTheme);
    const classes = useStyles();
    const [open, setOpen] = React.useState(false);

    function handleDrawerOpen() {
        setOpen(true);
    }

    function handleDrawerClose() {
        setOpen(false);
    }

    const initials = props.name.split(' ').map(word => word[0].toUpperCase());

    return (
        <div className={classes.root}>
            <CssBaseline />
            <AppBar
                color="primary"
                position="fixed"
                classes={{colorPrimary: appThemeContext.dark_mode ? classes.darkThemeAppbar : classes.liteThemeAppbar}}
            >
                <Toolbar className={classes.toolbar}>
                    <IconButton
                        color="inherit"
                        aria-label="Open drawer"
                        onClick={handleDrawerOpen}
                        edge="start"
                    >
                        <MenuIcon />
                    </IconButton>
                    <BannerLogo />
                    <div className={classes.padRight}></div>
                </Toolbar>
                <ProgressBar percent={33}/>
            </AppBar>
            <Drawer
                anchor="left"
                open={open}
                onClose={handleDrawerClose}
            >
                <div
                    className={classes.list}
                    role="presentation"
                >
                    <Box className={classes.accountDetails} >
                        <Box display="flex">
                            <Box className={classes.initialsContainer}>{initials}</Box>
                            <Box display="flex" flexDirection="column">
                                <Box>
                                    <Bold fontSize={18}>{props.name}</Bold>
                                </Box>
                                <Box>
                                    {props.email}
                                </Box>
                            </Box>
                        </Box>
                        <Box>
                            <Link to={ROUTES.ACCOUNT}>Account Details</Link>
                        </Box>
                    </Box>
                    <Divider />
                    <NavStepper onRouteClicked={handleDrawerClose}/>
                    <Divider />
                </div>
            </Drawer>
            <main>
                <div className={classes.drawerHeader} />
                <div className={drawerContent}>
                    {props.children}
                </div>
            </main>
        </div>
    );
}

export default PersistentDrawerLeft;
