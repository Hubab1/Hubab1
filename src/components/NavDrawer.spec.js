import React from 'react';
import { shallow } from 'enzyme';
import { PersistentDrawerLeft } from 'components/NavDrawer';

let defaultProps;
beforeEach(() => {
    defaultProps = {
        logout: jest.fn(),
        history: {
            push: jest.fn(),
            goBack: jest.fn(),
            location: {
                pathname: '',
            },
        },
        applicant: {
            first_name: 'John',
            last_name: 'Doe',
            email: 'jdoe@nest.io',
        },
        children: [],
        canAccessRoute: jest.fn(),
    };
});

it('PersistentDrawerLeft - Matches snapshot', function () {
    const wrapper = shallow(<PersistentDrawerLeft {...defaultProps} />);
    expect(wrapper.getElement()).toMatchSnapshot();
});
