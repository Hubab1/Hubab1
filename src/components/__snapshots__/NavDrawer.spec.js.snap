// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersistentDrawerLeft - Matches snapshot 1`] = `
<div>
  <ForwardRef(WithStyles) />
  <ForwardRef(WithStyles)
    className="makeStyles-appBar-3"
    color="primary"
    position="fixed"
  >
    <ForwardRef(WithStyles)
      className="makeStyles-toolbar-4"
    >
      <ForwardRef(WithStyles)
        aria-label="Open drawer"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <UNDEFINED />
      </ForwardRef(WithStyles)>
      <BannerLogo />
      <div
        className="makeStyles-avatarWrapper-12"
      >
        <div
          className="makeStyles-avatar-13"
        >
          J
          D
        </div>
        <ForwardRef(WithStyles)
          aria-controls={null}
          aria-haspopup={true}
          onClick={[Function]}
          size="small"
        >
          <UNDEFINED />
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          anchorEl={null}
          anchorOrigin={
            Object {
              "horizontal": "right",
              "vertical": "bottom",
            }
          }
          disableAutoFocus={false}
          disableEnforceFocus={false}
          disableRestoreFocus={false}
          id="menu"
          onClose={[Function]}
          onMouseLeave={[Function]}
          open={false}
        >
          <ForwardRef(MenuList)
            className="makeStyles-menuList-5"
          >
            <ForwardRef(WithStyles)
              onClick={[Function]}
            >
              Account
            </ForwardRef(WithStyles)>
            <ForwardRef(WithStyles)
              className="makeStyles-menuListItemDivider-6"
            />
            <ForwardRef(WithStyles)
              onClick={[Function]}
            >
              Logout
            </ForwardRef(WithStyles)>
          </ForwardRef(MenuList)>
        </ForwardRef(WithStyles)>
      </div>
    </ForwardRef(WithStyles)>
    <ProgressBar
      percent={0}
    />
  </ForwardRef(WithStyles)>
  <ForwardRef(WithStyles)
    anchor="left"
    onClose={[Function]}
    open={false}
  >
    <div
      className="makeStyles-list-1"
      role="presentation"
    >
      <ForwardRef(StyledComponent)
        className="makeStyles-section-7 makeStyles-communitySection-8"
      >
        <span />
        <span>
          Unit undefined
        </span>
      </ForwardRef(StyledComponent)>
      <ForwardRef(StyledComponent)
        className="makeStyles-section-7"
      >
        <UNDEFINED
          handleDrawerClose={[Function]}
        />
      </ForwardRef(StyledComponent)>
      <ForwardRef(StyledComponent)
        className="makeStyles-section-7 makeStyles-footer-10"
      >
        <ForwardRef
          target="_blank"
          to="/privacy-policy"
        >
          Privacy
        </ForwardRef>
        <ForwardRef
          target="_blank"
          to="/terms"
        >
          Terms of Use
        </ForwardRef>
        <ForwardRef
          target="_blank"
          to="/faq"
        >
          FAQs
        </ForwardRef>
      </ForwardRef(StyledComponent)>
    </div>
  </ForwardRef(WithStyles)>
  <main>
    <div
      className="makeStyles-drawerHeader-2"
    />
    <div
      className="css-us5k2a"
    />
  </main>
</div>
`;
